### PR TITLE
NCS 2.6.0 rebase

### PR DESCRIPTION
**Pull request by jfischer-no**

### CODE REVIEW

File: modules/trusted-firmware-m/Kconfig.tfm.crypto_modules [hunk 1]
```diff
@@ -81,11 +81,12 @@ config TFM_CRYPTO_ASYM_SIGN_MODULE_ENABLED
 config TFM_CRYPTO_KEY_DERIVATION_MODULE_ENABLED
 	bool "KEY DERIVATION crypto module"
 	default y
-	depends on PSA_HAS_KEY_DERIVATION && NRF_SECURITY
+	depends on (PSA_HAS_KEY_DERIVATION || PSA_HAS_KEY_AGREEMENT) && NRF_SECURITY
 	help
 	  Enables the KEY_DERIVATION crypto module within the crypto partition.
 	  Unset this option if the functionality provided by 'crypto_key_derivation.c'
 	  is not used.
+	  Note that key agreement is under key derivation in the current implementation.
```
Comment (jfischer-no) [inline modules/trusted-firmware-m/Kconfig.tfm.crypto_modules:89]: @SebastianBoe @mswarowsky Please take a look at the tfm and mbedtls noups, this fixup commit can not be squashed without rework.

**Review by rlubos — APPROVED** [review]
Zero-diff, commits seem to be squashed nicely, LGTM

**Comment by cvinayak** [comment]
> Zero-diff, commits seem to be squashed nicely, LGTM

I did the below:
```
# cd zephyr
# git fetch git@github.com:jfischer-no/sdk-zephyr.git pr-rebase-v2.6.0
# git diff HEAD FETCH_HEAD
```

I have some differences related LTO and `nrf-regtool 5.1.0 REQUIRED` etc... 

my HEAD is 3912101710

@rlubos is there a different way to compare that you got no diff?

**Comment by cvinayak** [comment]
> > Zero-diff, commits seem to be squashed nicely, LGTM
> 
> I did the below:
> 
> ```
> # cd zephyr
> # git fetch git@github.com:jfischer-no/sdk-zephyr.git pr-rebase-v2.6.0
> # git diff HEAD FETCH_HEAD
> ```
> 
> I have some differences related LTO and `nrf-regtool 5.1.0 REQUIRED` etc...
> 
> my HEAD is [3912101](https://github.com/nrfconnect/sdk-zephyr/commit/3912101710c23bfa3b6a2c5ab6fdecb6043a7133)
> 
> @rlubos is there a different way to compare that you got no diff?

Compared now with https://github.com/nrfconnect/sdk-zephyr/commit/8d93a161567dcc525771f70692587e6a37947b71

No diff.

**Review by cvinayak — APPROVED** [review]

**Review by tmon-nordic — APPROVED** [review]

**Comment by tmon-nordic** [comment]
The current main, i.e. 8d93a161567dcc525771f70692587e6a37947b71 is not tagged. It must be tagged before this PR is merged.

**Comment by jfischer-no** [comment]
I had to merge a few more PRs, should be in sync with sdk-zephyr/main, please recheck.

**Review by anangl — APPROVED** [review]

**Pull request merged**